captum/attr/_core/perm_feature_importance.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+import torch
+
+from .feature_ablation import FeatureAblation
+
+
+def permute_feature(x, feature_mask):
+    n = x.size(0)
+    perm = torch.randperm(n)
+    no_perm = torch.arange(n)
+    while (perm == no_perm).all():
+        perm = torch.randperm(n)
+
+    out = x.clone()
+    feature_mask = feature_mask.squeeze(0)
+    for i, j in enumerate(perm):
+        out[i][feature_mask] = x[j][feature_mask]
+
+    return out
+
+
+class PermutationFeatureImportance(FeatureAblation):
+    def __init__(self, forward_func=None, perm_fn=permute_feature):
+        super().__init__(forward_func=forward_func)
+        self.perm_fn = perm_fn
+
+    def attribute(
+        self,
+        inputs=None,
```

Comment:
We assume that input must be provided in all attribution algorithms, meaning that we can have `inputs` without defaulting it to `None`